hastory-server/src/Data/Hastory/Server/Utils.hs

```diff
@@ -0,0 +1,33 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module Data.Hastory.Server.Utils where
+
+import Crypto.Hash (Digest, SHA256(..), hashWith)
+import qualified Data.ByteString as B
+import Data.Hastory.Types (Entry(..), ServerEntry(..), SyncRequest(..))
+import qualified Data.Text as T
+import qualified Data.Text.Encoding as T
+
+hashEntry :: Entry -> T.Text -> Digest SHA256
+hashEntry Entry {..} host = hashWith SHA256 (unifiedData :: B.ByteString)
+  where
+    unifiedData =
+      mconcat
+        [ T.encodeUtf8 entryText
+        , (hashPrepare . show) entryWorkingDir
```

Comment:
I think an entry is a `FilePath` or a `Path Abs Dir` (can't remember).
`FilePath` is a type synonym for string so you don't need to show it.
`Path Abs Dir` is a wrapper that you can unwrap using `fromAbsDir.